### PR TITLE
stop persisting rejected blocks on P-chain

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Works towards resolving #1623

## How this works

We only call `AddStatelessBlock` with accepted blocks, removing future rejected blocks from being persisted to memory or disk.

Note: existing rejected blocks on disk are still there, a future PR will be made to prune them.

## How this was tested

CI

### CODE REVIEW

File: vms/platformvm/blocks/executor/rejector.go [hunk 1]
```diff
@@ -78,6 +77,5 @@ func (r *rejector) rejectBlock(b blocks.Block, blockType string) error {
 		}
 	}
 
-	r.state.AddStatelessBlock(b, choices.Rejected)
 	return r.state.Commit()
```
Comment (StephenButtolph) [inline vms/platformvm/blocks/executor/rejector.go:80]: We aren't committing anything here anymore - so I think we can remove this as well.

File: vms/platformvm/state/state.go [hunk 2]
```diff
@@ -1486,11 +1488,11 @@ func (s *state) init(genesisBytes []byte) error {
 	return s.Commit()
 }
 
-func (s *state) AddStatelessBlock(block blocks.Block, status choices.Status) {
+func (s *state) AddStatelessBlock(block blocks.Block) {
```
Comment (abi87) [inline vms/platformvm/state/state.go:1491]: nit: should we err if the block status is not actually accepted?

Comment (StephenButtolph) [reply vms/platformvm/state/state.go:1491]: We don't have that information here.

**Review by StephenButtolph — APPROVED** [review]

**Review by darioush — APPROVED** [review]

File: vms/platformvm/state/state.go [hunk 3]
```diff
@@ -125,7 +125,9 @@ type State interface {
 	SetLastAccepted(blkID ids.ID)
 
 	GetStatelessBlock(blockID ids.ID) (blocks.Block, choices.Status, error)
-	AddStatelessBlock(block blocks.Block, status choices.Status)
+
+	// Invariant: [block] is an accepted block.
+	AddStatelessBlock(block blocks.Block)
```
Comment (abi87) [inline vms/platformvm/state/state.go:130]: nit: should we rename this AddStatelessAcceptedBlock for clarify? Maybe it's too long of name

**Review by abi87 — APPROVED** [review]

**Comment by abi87** [comment]
@dhrubabasu  also nit: we cannot yet remove the status from GetStatelessBlock right? Not until we purge rejected blocks

File: vms/platformvm/state/state.go (hunk 3 above)
Comment (StephenButtolph) [reply vms/platformvm/state/state.go:130]: I feel like this change would be reverted when we remove status from `GetStatelessBlock`

**Pull request merged**